File: example/Monica_SoilTemp/src/cs/Monica_SoilTemp/SoilTemperatureCompState.cs
using System;
using System.Collections.Generic;
public class SoilTemperatureCompState 
{
    private double[] _V = new double[noOfTempLayers];
    private double[] _B = new double[noOfTempLayers];
    private double[] _volumeMatrix = new double[noOfTempLayers];
    private double[] _volumeMatrixOld = new double[noOfTempLayers];
    private double[] _matrixPrimaryDiagonal = new double[noOfTempLayers];
    private double[] _matrixSecondaryDiagonal = new double[noOfTempLayersPlus1];
    private double[] _heatConductivity = new double[noOfTempLayers];
    private double[] _heatConductivityMean = new double[noOfTempLayers];
    private double[] _heatCapacity = new double[noOfTempLayers];
    private double[] _solution = new double[noOfTempLayers];
    private double[] _matrixDiagonal = new double[noOfTempLayers];
    private double[] _matrixLowerTriangle = new double[noOfTempLayers];
    private double[] _heatFlow = new double[noOfTempLayers];
    private double _soilSurfaceTemperature;
    private double[] _soilTemperature = new double[22];
    private double _noSnowSoilSurfaceTemperature;
    
        public SoilTemperatureCompState() { }
    
    
    public SoilTemperatureCompState(SoilTemperatureCompState toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    V = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { V[i] = toCopy.V[i]; }
    
    B = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { B[i] = toCopy.B[i]; }
    
    volumeMatrix = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { volumeMatrix[i] = toCopy.volumeMatrix[i]; }
    
    volumeMatrixOld = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { volumeMatrixOld[i] = toCopy.volumeMatrixOld[i]; }
    
    matrixPrimaryDiagonal = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { matrixPrimaryDiagonal[i] = toCopy.matrixPrimaryDiagonal[i]; }
    
    matrixSecondaryDiagonal = new double[noOfTempLayersPlus1];
            for (int i = 0; i < noOfTempLayersPlus1; i++)
            { matrixSecondaryDiagonal[i] = toCopy.matrixSecondaryDiagonal[i]; }
    
    heatConductivity = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { heatConductivity[i] = toCopy.heatConductivity[i]; }
    
    heatConductivityMean = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { heatConductivityMean[i] = toCopy.heatConductivityMean[i]; }
    
    heatCapacity = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { heatCapacity[i] = toCopy.heatCapacity[i]; }
    
    solution = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { solution[i] = toCopy.solution[i]; }
    
    matrixDiagonal = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { matrixDiagonal[i] = toCopy.matrixDiagonal[i]; }
    
    matrixLowerTriangle = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { matrixLowerTriangle[i] = toCopy.matrixLowerTriangle[i]; }
    
    heatFlow = new double[noOfTempLayers];
            for (int i = 0; i < noOfTempLayers; i++)
            { heatFlow[i] = toCopy.heatFlow[i]; }
    
    soilSurfaceTemperature = toCopy.soilSurfaceTemperature;
    soilTemperature = new double[22];
            for (int i = 0; i < 22; i++)
            { soilTemperature[i] = toCopy.soilTemperature[i]; }
    
    noSnowSoilSurfaceTemperature = toCopy.noSnowSoilSurfaceTemperature;
    }
    }
    public double[] V
        {
            get { return this._V; }
            set { this._V= value; } 
        }
    public double[] B
        {
            get { return this._B; }
            set { this._B= value; } 
        }
    public double[] volumeMatrix
        {
            get { return this._volumeMatrix; }
            set { this._volumeMatrix= value; } 
        }
    public double[] volumeMatrixOld
        {
            get { return this._volumeMatrixOld; }
            set { this._volumeMatrixOld= value; } 
        }
    public double[] matrixPrimaryDiagonal
        {
            get { return this._matrixPrimaryDiagonal; }
            set { this._matrixPrimaryDiagonal= value; } 
        }
    public double[] matrixSecondaryDiagonal
        {
            get { return this._matrixSecondaryDiagonal; }
            set { this._matrixSecondaryDiagonal= value; } 
        }
    public double[] heatConductivity
        {
            get { return this._heatConductivity; }
            set { this._heatConductivity= value; } 
        }
    public double[] heatConductivityMean
        {
            get { return this._heatConductivityMean; }
            set { this._heatConductivityMean= value; } 
        }
    public double[] heatCapacity
        {
            get { return this._heatCapacity; }
            set { this._heatCapacity= value; } 
        }
    public double[] solution
        {
            get { return this._solution; }
            set { this._solution= value; } 
        }
    public double[] matrixDiagonal
        {
            get { return this._matrixDiagonal; }
            set { this._matrixDiagonal= value; } 
        }
    public double[] matrixLowerTriangle
        {
            get { return this._matrixLowerTriangle; }
            set { this._matrixLowerTriangle= value; } 
        }
    public double[] heatFlow
        {
            get { return this._heatFlow; }
            set { this._heatFlow= value; } 
        }
    public double soilSurfaceTemperature
        {
            get { return this._soilSurfaceTemperature; }
            set { this._soilSurfaceTemperature= value; } 
        }
    public double[] soilTemperature
        {
            get { return this._soilTemperature; }
            set { this._soilTemperature= value; } 
        }
    public double noSnowSoilSurfaceTemperature
        {
            get { return this._noSnowSoilSurfaceTemperature; }
            set { this._noSnowSoilSurfaceTemperature= value; } 
        }
}

File: example/Monica_SoilTemp/src/java/Monica_SoilTemp/SoilTemperatureCompState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class SoilTemperatureCompState
{
    private Double [] V;
    private Double [] B;
    private Double [] volumeMatrix;
    private Double [] volumeMatrixOld;
    private Double [] matrixPrimaryDiagonal;
    private Double [] matrixSecondaryDiagonal;
    private Double [] heatConductivity;
    private Double [] heatConductivityMean;
    private Double [] heatCapacity;
    private Double [] solution;
    private Double [] matrixDiagonal;
    private Double [] matrixLowerTriangle;
    private Double [] heatFlow;
    private double soilSurfaceTemperature;
    private Double [] soilTemperature;
    private double noSnowSoilSurfaceTemperature;
    
    public SoilTemperatureCompState() { }
    
    public SoilTemperatureCompState(SoilTemperatureCompState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            V = new Double[toCopy.getV().length];
        for (int i = 0; i < toCopy.getV().length; i++)
        {
            V[i] = toCopy.getV()[i];
        }
            B = new Double[toCopy.getB().length];
        for (int i = 0; i < toCopy.getB().length; i++)
        {
            B[i] = toCopy.getB()[i];
        }
            volumeMatrix = new Double[toCopy.getvolumeMatrix().length];
        for (int i = 0; i < toCopy.getvolumeMatrix().length; i++)
        {
            volumeMatrix[i] = toCopy.getvolumeMatrix()[i];
        }
            volumeMatrixOld = new Double[toCopy.getvolumeMatrixOld().length];
        for (int i = 0; i < toCopy.getvolumeMatrixOld().length; i++)
        {
            volumeMatrixOld[i] = toCopy.getvolumeMatrixOld()[i];
        }
            matrixPrimaryDiagonal = new Double[toCopy.getmatrixPrimaryDiagonal().length];
        for (int i = 0; i < toCopy.getmatrixPrimaryDiagonal().length; i++)
        {
            matrixPrimaryDiagonal[i] = toCopy.getmatrixPrimaryDiagonal()[i];
        }
            matrixSecondaryDiagonal = new Double[toCopy.getmatrixSecondaryDiagonal().length];
        for (int i = 0; i < toCopy.getmatrixSecondaryDiagonal().length; i++)
        {
            matrixSecondaryDiagonal[i] = toCopy.getmatrixSecondaryDiagonal()[i];
        }
            heatConductivity = new Double[toCopy.getheatConductivity().length];
        for (int i = 0; i < toCopy.getheatConductivity().length; i++)
        {
            heatConductivity[i] = toCopy.getheatConductivity()[i];
        }
            heatConductivityMean = new Double[toCopy.getheatConductivityMean().length];
        for (int i = 0; i < toCopy.getheatConductivityMean().length; i++)
        {
            heatConductivityMean[i] = toCopy.getheatConductivityMean()[i];
        }
            heatCapacity = new Double[toCopy.getheatCapacity().length];
        for (int i = 0; i < toCopy.getheatCapacity().length; i++)
        {
            heatCapacity[i] = toCopy.getheatCapacity()[i];
        }
            solution = new Double[toCopy.getsolution().length];
        for (int i = 0; i < toCopy.getsolution().length; i++)
        {
            solution[i] = toCopy.getsolution()[i];
        }
            matrixDiagonal = new Double[toCopy.getmatrixDiagonal().length];
        for (int i = 0; i < toCopy.getmatrixDiagonal().length; i++)
        {
            matrixDiagonal[i] = toCopy.getmatrixDiagonal()[i];
        }
            matrixLowerTriangle = new Double[toCopy.getmatrixLowerTriangle().length];
        for (int i = 0; i < toCopy.getmatrixLowerTriangle().length; i++)
        {
            matrixLowerTriangle[i] = toCopy.getmatrixLowerTriangle()[i];
        }
            heatFlow = new Double[toCopy.getheatFlow().length];
        for (int i = 0; i < toCopy.getheatFlow().length; i++)
        {
            heatFlow[i] = toCopy.getheatFlow()[i];
        }
            this.soilSurfaceTemperature = toCopy.getsoilSurfaceTemperature();
            soilTemperature = new Double[22];
        for (int i = 0; i < 22; i++)
        {
            soilTemperature[i] = toCopy.getsoilTemperature()[i];
        }
            this.noSnowSoilSurfaceTemperature = toCopy.getnoSnowSoilSurfaceTemperature();
        }
    }
    public Double [] getV()
    { return V; }

    public void setV(Double [] _V)
    { this.V= _V; } 
    
    public Double [] getB()
    { return B; }

    public void setB(Double [] _B)
    { this.B= _B; } 
    
    public Double [] getvolumeMatrix()
    { return volumeMatrix; }

    public void setvolumeMatrix(Double [] _volumeMatrix)
    { this.volumeMatrix= _volumeMatrix; } 
    
    public Double [] getvolumeMatrixOld()
    { return volumeMatrixOld; }

    public void setvolumeMatrixOld(Double [] _volumeMatrixOld)
    { this.volumeMatrixOld= _volumeMatrixOld; } 
    
    public Double [] getmatrixPrimaryDiagonal()
    { return matrixPrimaryDiagonal; }

    public void setmatrixPrimaryDiagonal(Double [] _matrixPrimaryDiagonal)
    { this.matrixPrimaryDiagonal= _matrixPrimaryDiagonal; } 
    
    public Double [] getmatrixSecondaryDiagonal()
    { return matrixSecondaryDiagonal; }

    public void setmatrixSecondaryDiagonal(Double [] _matrixSecondaryDiagonal)
    { this.matrixSecondaryDiagonal= _matrixSecondaryDiagonal; } 
    
    public Double [] getheatConductivity()
    { return heatConductivity; }

    public void setheatConductivity(Double [] _heatConductivity)
    { this.heatConductivity= _heatConductivity; } 
    
    public Double [] getheatConductivityMean()
    { return heatConductivityMean; }

    public void setheatConductivityMean(Double [] _heatConductivityMean)
    { this.heatConductivityMean= _heatConductivityMean; } 
    
    public Double [] getheatCapacity()
    { return heatCapacity; }

    public void setheatCapacity(Double [] _heatCapacity)
    { this.heatCapacity= _heatCapacity; } 
    
    public Double [] getsolution()
    { return solution; }

    public void setsolution(Double [] _solution)
    { this.solution= _solution; } 
    
    public Double [] getmatrixDiagonal()
    { return matrixDiagonal; }

    public void setmatrixDiagonal(Double [] _matrixDiagonal)
    { this.matrixDiagonal= _matrixDiagonal; } 
    
    public Double [] getmatrixLowerTriangle()
    { return matrixLowerTriangle; }

    public void setmatrixLowerTriangle(Double [] _matrixLowerTriangle)
    { this.matrixLowerTriangle= _matrixLowerTriangle; } 
    
    public Double [] getheatFlow()
    { return heatFlow; }

    public void setheatFlow(Double [] _heatFlow)
    { this.heatFlow= _heatFlow; } 
    
    public double getsoilSurfaceTemperature()
    { return soilSurfaceTemperature; }

    public void setsoilSurfaceTemperature(double _soilSurfaceTemperature)
    { this.soilSurfaceTemperature= _soilSurfaceTemperature; } 
    
    public Double [] getsoilTemperature()
    { return soilTemperature; }

    public void setsoilTemperature(Double [] _soilTemperature)
    { this.soilTemperature= _soilTemperature; } 
    
    public double getnoSnowSoilSurfaceTemperature()
    { return noSnowSoilSurfaceTemperature; }

    public void setnoSnowSoilSurfaceTemperature(double _noSnowSoilSurfaceTemperature)
    { this.noSnowSoilSurfaceTemperature= _noSnowSoilSurfaceTemperature; } 
    
}